primitives/api/proc-macro/src/impl_runtime_apis.rs

```diff
@@ -448,6 +493,100 @@ fn extend_with_api_version(mut trait_: Path, version: Option<u64>) -> Path {
 	trait_
 }
 
+// Adds a `#[cfg( feature = ...)]` attribute to attributes. Parameters:
+// feature_name - name of the feature
+// t - positive or negative feature guard.
+//		true => `#[cfg(feature = ...)]`
+//		false => `#[cfg(not(feature = ...))]`
+fn add_feature_guard(attrs: &mut Vec<Attribute>, feature_name: &str, t: bool) {
+	let attr = match t {
+		true => parse_quote!(#[cfg(feature = #feature_name)]),
+		false => parse_quote!(#[cfg(not(feature = #feature_name))]),
+	};
+	attrs.push(attr);
+}
+
+// Fold implementation which processes function implementations in impl block for a trait. It is
+// used to process `cfg_attr`s related to staging methods (e.g. `#[cfg_attr(feature =
+// "enable-staging-api", api_version(99))]`). Replaces the `cfg_attr` attribute with a feature guard
+// so that staging methods are added only for staging api implementations.
+struct ApiImplItem<'a> {
+	// Version of the trait where the `ImplItemFn` is located. Used for error reporting.
+	trait_api_ver: &'a Option<(String, u64)>,
+	// Span of the trait where the `ImplItemFn` is located. Used for error reporting.
+	trait_span: Span,
+	// All errors which occurred during folding
+	errors: Vec<Error>,
+}
+
+impl<'a> ApiImplItem<'a> {
+	pub fn new(trait_api_ver: &Option<(String, u64)>, trait_span: Span) -> ApiImplItem {
+		ApiImplItem { trait_api_ver, trait_span, errors: Vec::new() }
+	}
+
+	pub fn process(&mut self, item: ItemImpl) -> Result<ItemImpl> {
+		let res = self.fold_item_impl(item);
+
+		if let Some(e) = self.errors.pop() {
+			// TODO: what to do with the rest of the errors?
```

Comment:
I'm open for suggestions what to do in case there are multiple errors here. Atm just the first one is returned, which is okay-ish.
The alternative is to combine them, but I think it will look ugly.

Reply:
If we combine them, they will all look as 'one compilation error' on 4 different places, right? Won't this look weird?

Reply:
I've just tested it - it's not bad at all. I'll combine them.